### PR TITLE
add property details

### LINKED ISSUE #883: Property details
These three input groups (https://github.com/helpwave/web/issues/882) are all part of the property details layout. This layout should accept property data to be edited via these input groups. All changes are auto-saved.

The Field-Section Field-Type-Selector can just contain "Text" option right now. Later types will follow soon.

https://www.figma.com/file/4oxQRRLaNQ1TdBVhI0CZ4B/helpwave-frontend?type=design&node-id=3059-1578&mode=design&t=WISz1uPxaiY2qaH1-4
Image

### LINKED ISSUE #905: Property List View
### Describe the feature

A list for showing all properties in an organization, which can be sorted by the target of the property

### Why is this feature needed?

To use the properti s feature

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DasProffi**
## Which issues does this pull request close?
closes #883
closes #905

## [OPTIONAL] Give testing instructions to reviewers
1. Open the tasks page
2. Look at `/properties`

### CODE REVIEW

**Comment by MaxSchaefer** [comment]
On my To-do for today (06.03)

File: tasks/components/layout/property/PropertyDetails.tsx
Comment (MaxSchaefer) [inline tasks/components/layout/property/PropertyDetails.tsx:1]: Archive property button
Bildschirmfoto 2024-03-06 um 9 26 41 PM

File: tasks/components/layout/property/PropertyDetailsField.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  onChange
+}: PropsWithLanguage<PropertyDetailsFieldTranslation, PropertyDetailsFieldProps>) => {
+  const translation = useTranslation(language, defaultPropertyDetailsFieldTranslation)
+  return (
+    <InputGroup title={translation.field}>
+      <Select
+        // TODO add icons
+        value={value.fieldType}
+        label={{ name: translation.fieldType, labelType: 'labelMedium' }}
+        options={fieldTypeList.map(fieldType => ({ value: fieldType, label: translation[fieldType] }))}
```
Comment (MaxSchaefer) [inline tasks/components/layout/property/PropertyDetailsField.tsx:79]: Same translation "issue" as mentioned in `PropertySubjectTypeSelect`

File: tasks/components/layout/property/PropertyDetailsField.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      {(value.fieldType === 'multiSelect' || value.fieldType === 'singleSelect') && (
+        <div className={tw('flex flex-col mt-2 gap-y-1')}>
+          <div className={tw('flex flex-row justify-between items-center')}>
+            <Span type="labelMedium">{translation.values}</Span>
+            <Plus
+              className={tw('text-white bg-hw-primary-400 hover:text-gray-100 hover:bg-hw-primary-600 rounded-full mr-3')}
+              size={20}
+              onClick={() => {
+                const newList = [...value.entryList]
+                newList.push(translation.newEntry)
```
Comment (MaxSchaefer) [inline tasks/components/layout/property/PropertyDetailsField.tsx:91]: `translation.newEntry` is not a correct value to push. If you want to display `New entry`, use `placeholder="..."`. An empty value in this list should really be empty/null.

File: tasks/components/layout/property/PropertyDetailsField.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                onChange({ ...value, entryList: newList })
+              }}
+            />
+          </div>
+          <Scrollbars autoHide autoHeight autoHeightMax={400}>
+            <div className={tw('flex flex-col gap-y-2 mr-3')}>
+              {value.entryList.map((entry, index) => (
+                <div key={index} className={tw('flex flex-row items-center justify-between gap-x-4')}>
+                  <Input
```
Comment (MaxSchaefer) [inline tasks/components/layout/property/PropertyDetailsField.tsx:100]: As mentioned above. Use `placeholder="..."` for displaying `New entry`

File: tasks/components/layout/property/PropertyDetailsField.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        description={{ value: translation.allowCustomValuesDescription }}
+        suffix={(
+          <Checkbox
+            checked={value.isAllowingCustomValues}
+            onChange={isAllowingCustomValues => onChange({ ...value, isAllowingCustomValues })}
+            size={20}
+          />
+        )}
+        className={tw('mt-4')}
+      />
```
Comment (MaxSchaefer) [inline tasks/components/layout/property/PropertyDetailsField.tsx:138]: Also a part of multi-/single-selects

File: tasks/components/layout/property/PropertyDetailsField.tsx
Comment (MaxSchaefer) [inline tasks/components/layout/property/PropertyDetailsField.tsx:1]: fyi: In the future, this component could also contain an example representation of the resulting property type as displayed on the subjects details.

File: tasks/components/layout/property/PropertyDetailsRules.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  const translation = useTranslation(language, defaultPropertyDetailsRulesTranslation)
+  return (
+    <InputGroup title={translation.rules}>
+      <Select
+        // TODO add icons
+        value={value.importance}
+        label={{ name: translation.importance, labelType: 'labelMedium' }}
+        options={importanceList.map(importance => ({ value: importance, label: translation[importance] }))}
+        onChange={importance => onChange({ ...value, importance })}
+      />
```
Comment (MaxSchaefer) [inline tasks/components/layout/property/PropertyDetailsRules.tsx:62]: Soft-Required will be a checkbox as similar to always visible. You can use `Gets visually highlighted on an entity if the property is empty.`

File: tasks/components/layout/property/PropertyDisplay.tsx [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+        </div>
+      </div>
+      <Table
+        data={filteredProperties}
+        identifierMapping={dataObject => dataObject.id}
+        rowMappingToCells={property => [
+          (<Tile
+            key="cell-1"
+            title={{ value: property.basicInfo.propertyName }}
+            description={{ value: translation[property.field.fieldType] }}
```
Comment (MaxSchaefer) [inline tasks/components/layout/property/PropertyDisplay.tsx:228]: Same "translation"-Issue as already mentioned

File: tasks/components/layout/property/PropertyDisplay.tsx [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+            </div>
+          </Button>
+        </div>
+      </div>
+      <Table
+        data={filteredProperties}
+        identifierMapping={dataObject => dataObject.id}
+        rowMappingToCells={property => [
+          (<Tile
+            key="cell-1"
```
Comment (MaxSchaefer) [inline tasks/components/layout/property/PropertyDisplay.tsx:226]:
```suggestion
            key="field-type-cell"
```

File: tasks/components/layout/property/PropertyDisplay.tsx [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+            title={{ value: property.basicInfo.propertyName }}
+            description={{ value: translation[property.field.fieldType] }}
+          />),
+          (<div key="cell-2" className={tw('flex flex-row gap-x-2')}>
```
Comment (MaxSchaefer) [inline tasks/components/layout/property/PropertyDisplay.tsx:230]:
```suggestion
          (<div key={`subject-type-cell`} className={tw('flex flex-row gap-x-2')}>
```

File: tasks/components/layout/property/PropertyDisplay.tsx [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+            <SubjectTypeIcon subjectType={property.basicInfo.subjectType}/>
+            <Span>{translation[property.basicInfo.subjectType]}</Span>
+          </div>),
+          (<div key="cell-3" className={tw('flex flex-row justify-end')}>
```
Comment (MaxSchaefer) [inline tasks/components/layout/property/PropertyDisplay.tsx:234]:
```suggestion
          (<div key="edit-button-cell" className={tw('flex flex-row justify-end')}>
```
Do we really need all these keys?

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]
Pretty nice! Love to see this feature comes together! But still many things to to 👀
Some fields are not displayed during edit but I see `/properties` as a playground for integration right now. Therefore, lets not merge this against `main` but against `integration/properties`. We will consolidate approved changes on this pull request (I will create the corresponding branch protection).

For now, review the suggested changes please and we coordinate the first merge against `integration/properties` together. Great work! 🙌

File: lib/components/layout/Tile.tsx
[the remark this answers is not in the document]
Comment (DasProffi) [reply lib/components/layout/Tile.tsx:1]: The `Card` is the layouting around the content and the click handling, the `Tile` would be used for the inside to order components with a title, description, prefix, suffix etc.

File: tasks/components/layout/property/PropertySubjectTypeSelect.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  language,
+  ...props
+}: PropsWithLanguage<PropertySubjectTypeSelectTranslation, Omit<SelectProps<SubjectType>, 'options'>>) => {
+  const translation = useTranslation(language, defaultPropertySubjectTypeSelectTranslation)
+  return (
+    <Select
+      {...props}
+      options={subjectTypeList.map(subjectType => ({
+        value: subjectType,
+        label: translation[subjectType]
```
[the remark this answers is not in the document]
Comment (DasProffi) [reply tasks/components/layout/property/PropertySubjectTypeSelect.tsx:47]: > Do you know any other parts in our codebase were this occurs already?

Not that I am aware of

**Review by MaxSchaefer — COMMENTED** [review]
LGTM! Translations will be an upcoming todo for our medical team.

**Review by MaxSchaefer — APPROVED** [review]
I switched the base branch and configured a branch ruleset. Ready for merge!

File: tasks/components/layout/property/PropertyDetailsField.tsx [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+            <Checkbox
+              checked={value.isAllowingCustomValues}
+              onChange={isAllowingCustomValues => onChange({ ...value, isAllowingCustomValues })}
+              size={20}
+            />
+          )}
+          className={tw('mt-4')}
+        />
+      )}
+    </InputGroup>
```
[the remark this answers is not in the document]
Comment (DasProffi) [reply tasks/components/layout/property/PropertyDetailsField.tsx:138]: That would have introduced which a large `<>...</>` bracketing which might not be as readable as this

**Comment by DasProffi** [comment]
@MaxSchaefer the status does not seem to pass/progress

**Pull request merged**